docs/contributing.md

```diff
@@ -25,8 +25,78 @@ Guidelines](https://opensource.google/conduct/).
 
 ## Contribution process
 
+## Steps
+
+0. <Optionally needed> Create a virtual environment:
+
+Optionally use a venv to set up and develop xpk. This is needed for Google
+internal xpk development from a cloudtop machine.
+
+```shell
+  # Create venv if needed.
+  ## One time step of creating the venv
+  VENV_DIR=~/venvp3
+  python3 -m venv $VENV_DIR
+  ## enter the venv.
+  source $VENV_DIR/bin/activate
+  ## Same steps as above.
+  git clone https://github.com/google/xpk.git
+  pip install .[dev]
+```
+
+1. Install developer tools including `pyink`, `pylint`, and `precommit` using
+
+```shell
+  git clone https://github.com/google/xpk.git
+  pip install .[dev]
+```
+
+2. Install git hook scripts.
+```shell
+cd $XPK_REPO_DIR
+pre-commit install
+# Optionally run against files
+pre-commit run --all-files
+```
+
+3. Write code and commit changes!
+```shell
+# 1. Code will be autoformatted on:
+git commit -m "My cool new feature"
+# 2. If code needs to be reformatted, check what changes were made, and add them to the commit.
+git add -p
+# 3. Run git commit again, which should successfully pass pre-commit checks.
+git commit -m "My cool new feature"
+
+
+# Manually format code by running:
+pyink .
```

Comment:
What about pylint command? Is that not required?

Reply:
This shows how to manually format if needed. The pre-commit check will autoformat for you when you run git commit ...